src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java

```diff
@@ -148,24 +171,46 @@
         });
         story.addStep(new Statement() {
             @Override public void evaluate() throws Throwable {
-                SemaphoreStep.success("cleanupAfterRestart/1", null);
                 WorkflowJob p = story.j.jenkins.getItemByFullName("p", WorkflowJob.class);
                 assertNotNull(p);
                 WorkflowRun b = p.getBuildByNumber(1);
                 assertNotNull(b);
-                while (b.isBuilding()) { // TODO JENKINS-26399 need utility in JenkinsRule
+                assertEquals(Collections.<String>emptySet(), grep(b.getRootDir(), secret));
+                SemaphoreStep.success("cleanupAfterRestart/1", null);
+                while (b.isBuilding()) { // TODO 1.607+ use waitForCompletion
                     Thread.sleep(100);
                 }
                 story.j.assertBuildStatusSuccess(b);
-                story.j.assertLogNotContains("s3cr3t", b);
+                story.j.assertLogNotContains(secret, b);
                 FilePath key = story.j.jenkins.getNode("myslave").getWorkspaceFor(p).child("key");
                 assertTrue(key.exists());
-                assertEquals("s3cr3t", key.readToString());
+                assertEquals(secret, key.readToString());
                 FilePath secretFiles = story.j.jenkins.getNode("myslave").getRootPath().child("secretFiles");
                 assertTrue(secretFiles.isDirectory());
                 assertEquals(Collections.emptyList(), secretFiles.list());
+                assertEquals(Collections.<String>emptySet(), grep(b.getRootDir(), secret));
             }
         });
     }
 
+    private static Set<String> grep(File dir, String text) throws IOException {
+        Set<String> matches = new TreeSet<String>();
+        grep(dir, text, "", matches);
+        return matches;
+    }
+    private static void grep(File dir, String text, String prefix, Set<String> matches) throws IOException {
+        File[] kids = dir.listFiles();
+        if (kids == null) {
+            return;
+        }
+        for (File kid : kids) {
+            String qualifiedName = prefix + kid.getName();
+            if (kid.isDirectory()) {
+                grep(kid, text, qualifiedName + "/", matches);
+            } else if (kid.isFile() && FileUtils.readFileToString(kid).contains(text)) {
+                matches.add(qualifiedName);
+            }
+        }
+    }
+
```

Comment:
Seems like the above two `grep` methods could be useful utils to make available from the core test-harness.